mastodon.go

```diff
@@ -98,11 +99,28 @@ func (c *Client) doAPI(ctx context.Context, method string, uri string, params in
 		req.Header.Set("Content-Type", ct)
 	}
 
-	resp, err := c.Do(req)
-	if err != nil {
-		return err
+	var resp *http.Response
+	backoff := 1000 * time.Millisecond
+	for {
+		resp, err = c.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+
+		// handle status code 429, which indicates the server is throttling
+		// our requests. Do an exponential backoff and retry the request.
+		if resp.StatusCode == 429 {
+			if backoff > time.Hour {
+				break
+			}
+
+			backoff *= 2
+			time.Sleep(backoff)
```

Comment:
Should check ctx.Done() ?
```go
select {
case time.After(backoff)
case <-ctx.Done():
	return ctx.Err()
}
```

Reply:
Sorry this took so long, I must have accidentally deleted the GitHub notification.

Just pushed pretty much what you proposed, which handles the context cancellation.